lib/routes/page.js

```diff
@@ -889,65 +889,96 @@ module.exports = function(crowi, app) {
    * @apiParam {String} new_path
    * @apiParam {Bool} create_redirect
    */
-  api.rename = function(req, res) {
-    var pageId = req.body.page_id
-    var previousRevision = req.body.revision_id || null
-    var newPagePath = Page.normalizePath(req.body.new_path)
-    var newPageIsPortal = newPagePath.endsWith('/')
-    var options = {
-      createRedirectPage: (!newPageIsPortal && req.body.create_redirect) || 0,
-      moveUnderTrees: req.body.move_trees || 0,
+  api.rename = async function(req, res) {
+    const {
+      page_id: pageId,
+      revision_id: previousRevision = null,
+      new_path: newPath,
+      create_redirect: createRedirect,
+      move_trees: moveTrees,
+    } = req.body
+    const newPagePath = Page.normalizePath(newPath)
+    const newPageIsPortal = newPagePath.endsWith('/')
+    const options = {
+      createRedirectPage: (!newPageIsPortal && createRedirect) || 0,
+      moveUnderTrees: moveTrees || 0,
     }
 
     if (!Page.isCreatableName(newPagePath)) {
       return res.json(ApiResponse.error(sprintf('このページ名は作成できません (%s)', newPagePath)))
     }
 
-    var rename = function() {
-      var page = {}
-      Page.findPageById(pageId)
-        .then(function(pageData) {
-          page = pageData
-          if (!pageData.isUpdatable(previousRevision)) {
-            throw new Error("Someone could update this page, so couldn't delete.")
-          }
+    const rename = async function() {
+      try {
+        const page = await Page.findPageById(pageId)
+        if (!page.isUpdatable(previousRevision)) {
+          return res.json(ApiResponse.error(new Error("Someone could update this page, so couldn't delete.")))
+        }
+        const result = {
+          page: await Page.rename(page, newPagePath, req.user, options),
```

Comment:
This change of `result.page` causes an error on rename page action.

Originally, `result.page` was page object itsef,
https://github.com/crowi/crowi/pull/301/files#diff-db7e2375060f4b1d51ec816b5d6035bbL919

but this result of `await Page.rename` is the result of update of mongodb.

Reply:
I reverted this change 🙇